src/main/java/com/avispl/symphony/dal/communicator/aggregator/GoogleWorkspaceCommunicator.java

```diff
@@ -1037,9 +1086,19 @@ private JsonNode getJsonNodeByDeviceId(JsonNode jsonNode, String id) {
 	 * Otherwise, it returns the name of the organization unit at index 4 in the orgUnitList.
 	 */
 	private String getCurrentOrgUnitName() {
+		orgUnitNameList.clear();
 		if (StringUtils.isNotNullOrEmpty(filterOrgUnit)) {
 			return filterOrgUnit;
 		}
+		if (StringUtils.isNotNullOrEmpty(filterSerialNumber) && aggregatedDeviceResponse.size() > 0) {
+			for (JsonNode item : aggregatedDeviceResponse) {
+				orgUnitNameList.add(item.get(GoogleWorkspaceConstant.ORG_UNIT_PATH).asText().substring(1));
+			}
+			if (StringUtils.isNotNullOrEmpty(currentOrgUnitName)) {
+				return currentOrgUnitName;
+			}
+			return aggregatedDeviceResponse.get(GoogleWorkspaceConstant.DEFAULT_ORG_UNIT_POSITION).get(GoogleWorkspaceConstant.ORG_UNIT_PATH).asText().substring(1);
```

Comment:
get(GoogleWorkspaceConstant.DEFAULT_ORG_UNIT_POSITION) and get(GoogleWorkspaceConstant.ORG_UNIT_PATH) are always gonna be here, right? Just worried about potential NPEs